pyomo/repn/plugins/gams_writer.py

```diff
@@ -249,23 +376,155 @@ def _print_model( self, model, output_file, solver_capability, var_list,
         output_file.write(";\n\n")
         output_file.write(ConstraintIO.getvalue())
         output_file.write("\n")
-        for varName in positive:
+        for varName in var_list:
             var = symbolMap.getObject(varName)
-            if var.ub is not None:
-                output_file.write("%s.up = %s;\n" % (varName, var.ub))
-        for varName in ints + reals:
-            var = symbolMap.getObject(varName)
-            if var.lb is not None:
-                output_file.write("%s.lo = %s;\n" % (varName, var.lb))
-            if var.ub is not None:
-                output_file.write("%s.up = %s;\n" % (varName, var.ub))
-
-        model_name = symbolMap.getSymbol(model, labeler)
-        output_file.write("\nMODEL %s /all/ ;\n"
-                          % model_name )
+            if var.is_expression():
+                continue
+            if varName in positive:
+                if var.ub is not None:
+                    output_file.write("%s.up = %s;\n" % (varName, var.ub))
+            elif varName in posInts:
+                if var.lb != 0:
+                    output_file.write("%s.lo = %s;\n" % (varName, var.lb))
+                if var.ub is not None:
+                    output_file.write("%s.up = %s;\n" % (varName, var.ub))
+            elif varName in otherInts:
+                if var.lb is None:
+                    # GAMS doesn't allow -INF lower bound for ints
+                    # Set bound to lowest possible bound in Gams
+                    output_file.write("%s.lo = -1.0E+10;\n" % (varName))
```

Comment:
Agreed.  We got burned a while back when a solver "guessed" a bound when one wasn't provided.